src/cmd_all/src/single_node.rs

```diff
@@ -96,50 +139,282 @@ pub fn make_single_node_state_store_url(store_directory: &String) -> String {
 }
 
 pub fn map_single_node_opts_to_standalone_opts(opts: &SingleNodeOpts) -> ParsedStandaloneOpts {
-    let mut meta_opts = SingleNodeOpts::default_meta_opts();
-    let mut compute_opts = SingleNodeOpts::default_compute_opts();
-    let mut frontend_opts = SingleNodeOpts::default_frontend_opts();
-    let mut compactor_opts = SingleNodeOpts::default_compactor_opts();
+    todo!()
+}
+
+pub fn normalized_single_node_opts(opts: &SingleNodeOpts) -> ParsedStandaloneOpts {
+    let mut meta_opts = RawOpts::default_meta_opts();
+    let mut compute_opts = RawOpts::default_compute_opts();
+    let mut frontend_opts = RawOpts::default_frontend_opts();
+    let mut compactor_opts = RawOpts::default_compactor_opts();
```

Comment:
So this is the core idea: We now first create `RawOpts` (essentially string-like opt), update it, and finally create `Opts` in one shot. 

The previous approach (#15044) is create `Opts` from the beginning, and use `clap::Parser::update_from` to update it, which has some limitations about the default values (although I'm not very familiar with that).

Do I understand it correctly? If so, it looks good to me. It looks just a subtle implementation detail change to me.

Reply:
> If so, it looks good to me. It looks just a subtle implementation detail change to me.

I mean I don't care too much about the implementation detail, as long as it works.. (Disclaimer: I didn't think too much about whether this PR's implementation is correct 🤪)